client/solve.go

```diff
@@ -5,15 +5,29 @@ import (
 	"crypto/rand"
 	"encoding/hex"
 	"io"
+	"os"
+	"path/filepath"
+	"strings"
 	"time"
 
+	"github.com/Sirupsen/logrus"
 	controlapi "github.com/moby/buildkit/api/services/control"
 	"github.com/moby/buildkit/client/llb"
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/session/filesync"
+	"github.com/moby/buildkit/session/grpchijack"
+	"github.com/moby/buildkit/solver/pb"
 	"github.com/pkg/errors"
 	"golang.org/x/sync/errgroup"
 )
 
-func (c *Client) Solve(ctx context.Context, r io.Reader, statusChan chan *SolveStatus, exporter string, exporterAttrs map[string]string) error {
+func (c *Client) Solve(ctx context.Context, r io.Reader, statusChan chan *SolveStatus, exporter string, exporterAttrs map[string]string, localDir string) error {
```

Comment:
It is hard to follow a function with more than 100 lines :sweat_smile: 
(Can be another PR though)